libdnf/dnf-sack.cpp

```diff
@@ -526,22 +545,13 @@ write_main(DnfSack *sack, HyRepo hrepo, int switchtosolv, GError **error)
         }
     }
     if (switchtosolv && repo_is_one_piece(repo)) {
+        repo_empty(repo, 1);
         /* switch over to written solv file activate paging */
-        FILE *fp = fopen(tmp_fn_templ, "r");
-        if (fp) {
-            repo_empty(repo, 1);
-            rc = repo_add_solv(repo, fp, 0);
-            fclose(fp);
-            if (rc) {
-                /* this is pretty fatal */
-                ret = FALSE;
-                g_set_error_literal (error,
-                                     DNF_ERROR,
-                                     DNF_ERROR_FILE_INVALID,
-                                     _("write_main() failed to re-load "
-                                       "written solv file"));
-                goto done;
-            }
+        gboolean loaded = try_to_use_cached_solvfile(tmp_fn_templ, repo, 0, repoImpl->checksum, error);
+        if (!loaded || (error && *error)) {
+            /* this is pretty fatal, we should be able to use a file we just wrote */
+            ret = FALSE;
+            goto done;
```

Comment:
The code previously didn't re-check the checksum. Now if the load fails due to a checksum mismatch (for whatever reason), the error won't be set and this again fails silently... (FWIW I also find the "pretty fatal" comment quite useless :slightly_smiling_face: )

Reply:
Added additional error handling for that.